nodejs/awsx/elasticloadbalancingv2/loadBalancer.ts

```diff
@@ -138,3 +153,52 @@ export interface LoadBalancerSubnets {
 function isLoadBalancerSubnets(obj: any): obj is LoadBalancerSubnets {
     return obj && (<LoadBalancerSubnets>obj).subnets instanceof Function;
 }
+
+export interface LoadBalancerTarget {
+    /**
+     * The ID of the target. This is the Instance ID for an `instance`, or the container ID for an
+     * ECS container. If the target type is `ip`, specify an IP address. If the target type is
+     * `lambda`, specify the arn of lambda.
+     */
+    targetId: string;
+    /**
+     * The Availability Zone where the IP address of the target is to be registered.
+     */
+    availabilityZone?: string;
+    /**
+     * The port on which targets receive traffic.
+     */
+    port?: number;
+}
+
+export interface LoadBalancerTargetProvider {
+    loadBalancerTarget(targetType: pulumi.Input<mod.TargetType>): pulumi.Output<LoadBalancerTarget>;
+}
+
+export function isLoadBalancerTargetProvider(obj: any): obj is LoadBalancerTargetProvider {
+    return (<LoadBalancerTargetProvider>obj).loadBalancerTarget instanceof Function;
+}
+
+/**
+ * Allows an EC2 instance to simply be used as the target of an ALB or NLB.  To use, just call:
+ *
+ * ```ts
+ *  lb.attachTarget(new Ec2InstanceTarget(instance));
+ * ```
+ */
+export class Ec2InstanceTarget implements LoadBalancerTargetProvider {
+    constructor(public readonly instance: aws.ec2.Instance) {
+    }
+
+    public loadBalancerTarget(targetType: pulumi.Input<mod.TargetType>): pulumi.Output<LoadBalancerTarget> {
+        const result = pulumi.output([targetType, this.instance.id, this.instance.privateIp, this.instance.availabilityZone])
+                             .apply(([targetType, instanceId, privateIp, availabilityZone]) => {
+            return {
+                targetId: targetType === "instance" ? instanceId : privateIp,
+                availabilityZone: availabilityZone,
```

Comment:
@joeduffy was looking but couldn't find good docs here.  Do you know of any?  It will help me determine if this behavior is the one we want for all cases, or if we need to expose some level of customization to control things more.

Reply:
Good docs on what exactly?

Reply:
I think https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html#target-type is the best. Note this part:

> When the target type is ip, you can specify IP addresses from one of the following CIDR blocks:
> 
> The subnets of the VPC for the target group
> 
> * 10.0.0.0/8 (RFC 1918)
> * 100.64.0.0/10 (RFC 6598)
> * 172.16.0.0/12 (RFC 1918)
> * 192.168.0.0/16 (RFC 1918)
> 
> **Important**
> You can't specify publicly routable IP addresses.

Reply:
yes, that's very helpful.  let me read through this to figure out what else may need to change.  thanks!